electrumx/lib/tx.py

```diff
@@ -101,7 +103,8 @@ def read_tx_and_hash(self):
         we process it in the natural serialized order.
         '''
         start = self.cursor
-        return self.read_tx(), double_sha256(self.binary[start:self.cursor])
+        return self.read_tx(), type(self).TX_HASH_FN(
```

Comment:
Why type(self) in these 3 lines?  just "self." should work fine.

I hope this doesn't have much impact on speed.

Reply:
John - you seem to be offline.  Thanks for the PR.
I'll make this change just using self.TX_HASH_FN which has your desired semantics and my tests show to run 75% faster in the interpreter

Reply:
Hmm, I'm seeing why you did this now, lol

Reply:
TX_HASH_FN = staticmethod(double_sha256) works; learn something every day.

I've pushed this.